lib/api/controllers/authController.js

```diff
@@ -7,26 +7,53 @@ module.exports = function AuthController (kuzzle) {
 
   this.passport = new PassportWrapper(kuzzle);
 
-  this.login = function (requestObject) {
+  this.logout = function (requestObject, context) {
+    var
+      deferred = q.defer(),
+      response;
+
+    kuzzle.pluginsManager.trigger('auth:logout', context)
+      .then(kuzzle.repositories.token.expire(context.token))
```

Comment:
This doesn't work.
According to [promises specifications](https://promisesaplus.com/) followed by Q, the `then` method takes two arguments: `onFulfilled` and `onRejected`.
If these arguments aren't functions, they are ignored.

What happens here is that `kuzzle.repositories.token.expire` is executed but its result is not considered for the next then()/catch() statements. So the execution of this call stack only depends on the `pluginsManager.trigger` statement.

This should be changed to: `.then(() => { return kuzzle.repositories.token.expire(context.token); })`